modules/standard/JSON.chpl

```diff
@@ -249,29 +434,77 @@ module JSON {
         _first = true;
       }
 
-      @chpldoc.nodoc
+      /*
+        Serialize ``element``.
+
+        Writes a leading comma if this is not the first element in the row.
+      */
       proc ref writeElement(const element: ?) throws {
         if !_first then writer._writeLiteral(", ");
         else _first = false;
 
         writer.write(element);
       }
 
-      @chpldoc.nodoc
+      /*
+        Ends serialization of the current array.
+      */
       proc endArray() throws {
       }
     }
 
-    @chpldoc.nodoc
-    proc startArray(writer: jsonWriter, size: int) throws {
-      return new ArraySerializer(writer);
+    /*
+      Start serializing a map by writing the character ``{``.
+
+      :arg writer: The ``fileWriter`` to be used when serializing.
+      :arg size: The number of entries in the map.
+
+      :returns: A new MapSerializer
+    */
+    proc startMap(writer: jsonWriter, size: int) throws {
+      writer._writeLiteral("{");
+      return new MapSerializer(writer);
     }
 
+    /*
+      Returned by ``startMap`` to provide the API for serializing maps.
+
+      Maps are serialized as JSON objects. For example, a map of strings to
+      strings would be serialized as:
+
+      .. code-block:: text
+
+        {
+          "east": "west",
+          "hello": "goodbye",
+          "north": "south",
+          "day": "night"
+        }
+
+      The indenting and whitespace of this example is not considered stable.
+
+    */
     record MapSerializer {
+      @chpldoc.nodoc
       var writer;
+      @chpldoc.nodoc
       var _first : bool = true;
 
-      @chpldoc.nodoc
+      /*
+        Serialize ``key``.
+
+        JSON itself only supports strings as names in objects. This library
```

Comment:
This seems like an explanation that could be in a `::note`, coupled with the above.